packages/api/src/brevo-contact/brevo-contact-import.service.ts

```diff
@@ -29,114 +46,151 @@ export class BrevoContactImportService {
-                    );
-                    if (updatedBrevoContact) updated++;
-                    else failed++;
-                } else if (!brevoContact) {
-                    const success = await this.brevoContactsService.createDoubleOptInContact({
-                        ...contact,
-                        scope,
-                        templateId: this.config.brevo.resolveConfig(scope).doubleOptInTemplateId,
-                        listIds: targetGroupBrevoIds,
-                    });
-                    if (success) created++;
-                    else failed++;
-                }
-            } catch (err) {
-                console.error(err);
+            } catch (validationError) {
+                console.error(validationError);
+                failedColumns.push(row);
                 failed++;
             }
         }
-        return { created, updated, failed };
+        if (created + updated + failed === 0) {
+            return { created, updated, failed, failedColumns, errorMessage: "No contacts found." };
+        }
+
+        return { created, updated, failed, failedColumns };
     }
 
-    async parseCsvToBrevoContacts(csvContent: string, redirectUrl: string): Promise<CreateDoubleOptInContactData[]> {
-        const brevoContacts: CreateDoubleOptInContactData[] = [];
-
-        return new Promise((resolve, reject) => {
-            parseString(csvContent, { headers: true, delimiter: ";" })
-                .on("error", (error) => {
-                    console.error(error);
-                    reject(error);
-                })
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                .on("data", async (row: Record<string, any>) => {
-                    try {
-                        const contactData = await this.processRow(row, redirectUrl);
-                        brevoContacts.push(contactData);
-                    } catch (validationError) {
-                        console.error(validationError);
-                        reject(validationError);
-                    }
-                })
-                .on("end", (rowCount: number) => {
-                    console.log(`Parsed ${rowCount} rows`);
-                    resolve(brevoContacts);
+    private async createOrUpdateBrevoContact(
+        contact: CreateDoubleOptInContactData,
+        scope: EmailCampaignScopeInterface,
+        targetGroupBrevoIds: number[],
+    ): Promise<"created" | "updated" | "error"> {
+        try {
+            let brevoContact;
+            try {
+                brevoContact = await this.brevoApiContactsService.findContact(contact.email, scope);
+            } catch (error) {
+                // Brevo throws 404 error if no contact was found
+                if (!isErrorFromBrevo(error)) {
+                    throw error;
+                }
+                if (error.response.statusCode !== 404) {
+                    throw error;
+                }
```

Comment:
The comment should be above the second if i think

Reply:
yes, thanks.

done in: 7254d7fd4219199315a3b0834f01a6a6537c6304